letsencrypt/tests/auth_handler_test.py

```diff
@@ -148,6 +155,37 @@ def test_perform_failure(self):
         self.assertRaises(
             errors.AuthorizationError, self.handler.get_authorizations, ["0"])
 
+    @mock.patch("acme.challenges.HTTP01Response.simple_verify")
+    @mock.patch("acme.challenges.TLSSNI01Response.simple_verify")
+    def test_simple_verify(self, mock_http01_verify, mock_tlssni01_verify):
+        # pylint: disable=protected-access
+        key = jose.JWK.load(test_util.load_vector('rsa512_key.pem'))
+        challenge_http_01 = achallenges.KeyAuthorizationAnnotatedChallenge(
+            challb=acme_util.HTTP01_P, domain=b'localhost', account_key=key)
+        challenge_tls_sni_01 = achallenges.KeyAuthorizationAnnotatedChallenge(
+            challb=acme_util.TLSSNI01_P, domain=b'localhost', account_key=key)
+
+        mock_http01_verify.return_value = False
+        mock_tlssni01_verify.return_value = False
+
+        response_http_01 = challenges.HTTP01Response(key_authorization=u'foo')
+        response_tls_sni_01 = challenges.TLSSNI01Response(key_authorization=u'foo')
+
+        self.handler.dv_c = [challenge_http_01, challenge_tls_sni_01]
+        self.mock_dv_auth.perform.side_effect = [[response_http_01, response_tls_sni_01]]
```

Comment:
I still don't understand why you cannot use `.return_value = [response_http_01, response_tls_sni_01]`...